src/main/kotlin/de/danielmaile/mpp/item/ItemRegistry.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * This file is part of MPP.
+ * Copyright (c) 2023 by it's authors. All rights reserved.
+ * MPP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MPP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MPP.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.danielmaile.mpp.item
+
+import de.danielmaile.mpp.util.getDataString
+import org.bukkit.inventory.ItemStack
+
+const val MPP_ITEM_TAG_KEY = "mpp_item"
+const val MPP_ITEM_PREFIX = "mpp"
+
+object ItemRegistry {
+    // id (mpp:name) to Item
+    private val items = HashMap<String, Item>()
+
+    fun registerItems(items: Array<out Item>) {
+        for (item in items) {
+            item.register()
+        }
+    }
+    fun getAllItems(): Array<out Item> {
+        return items.values.toTypedArray()
+    }
+
+    fun registerItem(item: Item) {
+        val id = "$MPP_ITEM_PREFIX:${item.name.lowercase()}"
+        registerItem(id, item)
+        item.registerRecipes()
+    }
+
+    fun registerItem(id: String, item: Item) {
+        if (items.containsKey(id) || items.containsValue(item)) {
+            throw IllegalArgumentException("Item with id: $id is already registered")
+        }
+
+        items[id] = item
+    }
+
+    fun getIdFromItemstack(itemStack: ItemStack): String? {
```

Comment:
We should use camel case like spigot does it. Same for the next few occurrences.
```suggestion
    fun getIdFromItemStack(itemStack: ItemStack): String? {
```